esp/platform/espcontext.cpp

```diff
@@ -512,28 +512,59 @@ class CEspContext : public CInterface, implements IEspContext
     }
 
     virtual void addTraceSummaryValue(LogLevel logLevel, const char *name, const char *value, const unsigned int group = TXSUMMARY_GRP_CORE)
+    {
+        addTraceSummaryValue(logLevel, name, value, nullptr, group);
+    }
+
+    virtual void addTraceSummaryValue(LogLevel logLevel, const char *name, const char *value, const char* otName, const unsigned int group = TXSUMMARY_GRP_CORE)
     {
         if (m_txSummary && !isEmptyString(name))
-            m_txSummary->append(name, value, logLevel, group);
+            m_txSummary->append(name, value, otName, TxUnits::NA, logLevel, group);
     }
 
     virtual void addTraceSummaryValue(LogLevel logLevel, const char *name, __int64 value, const unsigned int group = TXSUMMARY_GRP_CORE)
+    {
+        addTraceSummaryValue(logLevel, name, value, nullptr, TxUnits::NA, group);
+    }
+
+    virtual void addTraceSummaryValue(LogLevel logLevel, const char *name, __int64 value, const char* otName, unsigned group = TXSUMMARY_GRP_CORE) override
+    {
+        addTraceSummaryValue(logLevel, name, value, otName, TxUnits::NA, group);
+    }
+
+    virtual void addTraceSummaryValue(LogLevel logLevel, const char *name, __int64 value, TxUnits units, unsigned group = TXSUMMARY_GRP_CORE) override
+    {
+        addTraceSummaryValue(logLevel, name, value, nullptr, units, group);
+    }
+
+    virtual void addTraceSummaryValue(LogLevel logLevel, const char *name, __int64 value, const char* otName, TxUnits units, unsigned group = TXSUMMARY_GRP_CORE) override
     {
         if (m_txSummary && !isEmptyString(name))
-            m_txSummary->append(name, value, logLevel, group);
+            m_txSummary->append(name, value, otName, units, logLevel, group);
     }
 
     virtual void addTraceSummaryDoubleValue(LogLevel logLevel, const char *name, double value, const unsigned int group = TXSUMMARY_GRP_CORE)
+    {
+        addTraceSummaryDoubleValue(logLevel, name, value, nullptr, group);
+    }
+
+    virtual void addTraceSummaryDoubleValue(LogLevel logLevel, const char *name, double value, const char* otName, unsigned group = TXSUMMARY_GRP_CORE)
     {
         if (m_txSummary && !isEmptyString(name))
-            m_txSummary->append(name, value, logLevel, group);
+            m_txSummary->append(name, value, otName, TxUnits::NA, logLevel, group);
     }
 
     virtual void addTraceSummaryTimeStamp(LogLevel logLevel, const char *name, const unsigned int group = TXSUMMARY_GRP_CORE)
+    {
+        addTraceSummaryTimeStamp(logLevel, name, nullptr, group);
+    }
+
+    virtual void addTraceSummaryTimeStamp(LogLevel logLevel, const char *name, const char* otName, unsigned group = TXSUMMARY_GRP_CORE) override
     {
         if (m_txSummary && !isEmptyString(name))
-            m_txSummary->append(name, m_txSummary->getElapsedTime(), logLevel, group, "ms");
+            m_txSummary->append(name, m_txSummary->getElapsedTime(), otName, TxUnits::millis, logLevel, group, "ms");
```

Comment:
You are correct that we have not found a reason to use the templated serializer. It could removed, if we're comfortable changing an interface that has been available to external ESP developers for years.

The suffix is obsolete, at least to us. Our only use of it was to attach units to numbers. With the suffix attached to the value when appended, it caused invalid JSON numeric output.  It too could be removed.

Would it be better to consider these changes in a separate PR?

Reply:
One of my (possibly misguided) goals was to minimize the changes made to the existing trace summary interfaces. Based on the assumption that scapps might adopt the solution, I thought that keeping as much of the change as possible behind the existing interface would encourage adoption.

Scapps will not be adopting the solution. I believe they have taken the approach of adding new annotations for otel, leaving their TxSummary values and behavior unchanged. I'm wondering if we shouldn't do the same thing.

Most of the existing TxSummary instrumentation is millisecond measurements. Existing TxSummary analytics expect milliseconds. Changing the values to nanoseconds to satisfy otel either invalidates TxSummary output or requires scaling values back to milliseconds.

After some discussion with @rpastrana, it seems we'll consider replacing many of the timing values that represent start and end offsets with client spans that can't be represented in the TxSummary.

Is there still enough value to make a single interface that updates both otel and TxSummary?

If yes, should the inputs satisfy otel expectations, or TxSummary?

If inputs satisfy otel expectations, is maintaining current TxSummary expectations for the same values a requirement?

To be clear, I'm not arguing for any solution. Two reviewers have now made it clear that I completely misunderstood the problem statement. I need more guidance to know the problem I'm trying to solve.